test.js

```diff
@@ -1,71 +1,528 @@
-"use strict";
-var cache = require('./index');
+/* global describe, it, before, beforeEach, afterEach */
+'use strict';
 
-cache.debug(false);
+var chai = require('chai'),
+    expect = chai.expect,
+    sinon = require('sinon'),
+    sinonChai = require('sinon-chai'),
+    cache = require('./index'),
+    clock;
 
-cache.put('a', true);
-console.log('true == %s', cache.get('a'));
-cache.clear();
-console.log('null == %s', cache.get('a'));
+chai.use(sinonChai);
 
-console.log('null == %s', cache.get('a'));
-console.log('0 == %s', cache.size());
 
-cache.put('a', 'b', 3000);
-console.log('1 == %s', cache.size());
+describe('node-cache', function() {
+  beforeEach(function() {
+    clock = sinon.useFakeTimers();
 
-console.log('b == %s', cache.get('a'));
+    cache.clear();
+  });
 
-console.log("keys == ", cache.keys());
+  afterEach(function() {
+    clock.restore();
+  });
 
-var complicated = ['a',{'b':'c','d':['e',3]},'@'];
-cache.put(complicated, true);
-console.log('true == %s', cache.get(complicated));
-cache.del(complicated);
-console.log('null == %s', cache.get(complicated));
+  describe('put()', function() {
```

Comment:
You may be wondering where all the extensive tests for `put()` are. The trouble here is that we want to test all the methods in isolation, so we shouldn't use `get()`, `del()`, `size()`, etc. here since we don't know at this point in the test suite if they actually work. So we test what we can here and then in almost every other test, we put `put()`through its paces. So in the end, we do extensively test this method.